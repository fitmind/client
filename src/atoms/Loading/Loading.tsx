import React from 'react';
import { connect } from 'react-redux';
import { ApplicationState } from '../../redux/reducers/root.reducer';
import { Spinner } from 'react-bootstrap';
import styled from 'styled-components';

interface PropsFromState {
    loading: boolean;
}

const LoadingComponent: React.FC<PropsFromState> = ({ loading }) => (
    <div>
        {loading && (
            <Background>
                <Spinner animation="grow" variant="success">
                    <span className="sr-only">Loading...</span>
                </Spinner>
            </Background>
        )}
    </div>
);

const mapStateToProps = (state: ApplicationState) => ({
    loading: state.ui.loading,
});

export default connect(
    mapStateToProps,
    null,
)(LoadingComponent);

const Background = styled.div`
    background: rgba(0, 0, 0, 0.5); /* Green background with 30% opacity */
    color: white;
    height: 100vh;
    width: 100vw;
    position: absolute;
    left: 0;
    top: 0;
    z-index: 10;
    display: flex;
    align-items: center;
    justify-content: center;
`;
