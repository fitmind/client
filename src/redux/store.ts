import { Store, createStore, applyMiddleware } from 'redux';
import { ApplicationState, storeInitialState, createRootReducer } from './reducers/root.reducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import { rootSaga } from './saga/root.saga';
import { History, createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

// We use hash history because this example is going to be hosted statically.
// Normally you would use browser history.
export const history: History = createBrowserHistory();

export const configureStore = (initialState: ApplicationState): Store<ApplicationState> => {
    const sagaMiddleware = createSagaMiddleware();
    const composeEnhancers = composeWithDevTools({});
    let store = createStore(
        createRootReducer(history),
        initialState,
        composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware)),
    );

    sagaMiddleware.run(rootSaga);
    return store;
};

const store: Store<ApplicationState> = configureStore(storeInitialState);

export default store;
