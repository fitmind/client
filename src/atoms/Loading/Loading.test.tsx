import React from 'react';
import { shallow } from 'enzyme';
import store, { configureStore } from '../../redux/store';
import { Provider } from 'react-redux';
import Loading from './Loading';
import { storeInitialState } from '../../redux/reducers/root.reducer';
import { mergeDeepRight } from 'ramda';
import renderer from 'react-test-renderer';

describe('Loading Component', () => {
    it('renders without crashing', () => {
        shallow(
            <Provider store={store}>
                <Loading />
            </Provider>,
        );
    });
    it('should match the snapshot when the loading state is true', () => {
        const mergedProps = mergeDeepRight(storeInitialState, {
            ui: { loading: true },
        });
        const storeWithLoadingTrue = configureStore(mergedProps);
        const component = renderer.create(
            <Provider store={storeWithLoadingTrue}>
                <Loading />
            </Provider>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should match the snapshot when the loading state is fakse', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Loading />
            </Provider>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
