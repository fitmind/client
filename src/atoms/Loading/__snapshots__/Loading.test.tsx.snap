// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should match the snapshot when the loading state is fakse 1`] = `<div />`;

exports[`Loading Component should match the snapshot when the loading state is true 1`] = `
<div>
  <div
    className="sc-bdVaJa isMKoN"
  >
    <div
      className="spinner-grow text-success"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
